addons/stock/report/report_stock_forecasted.py

```diff
@@ -74,19 +74,15 @@ def _get_report_data(self, product_template_ids=False, product_variant_ids=False
         assert product_template_ids or product_variant_ids
         res = {}
 
-        # Get the warehouse we're working on as well as its locations.
-        if self.env.context.get('warehouse'):
+        if 'warehouse' in self.env.context:
             warehouse = self.env['stock.warehouse'].browse(self.env.context['warehouse'])
         else:
-            warehouse = self.env['stock.warehouse'].search([
-                ('company_id', '=', self.env.company.id)
-            ], limit=1)
-            self.env.context = dict(self.env.context, warehouse=warehouse.id)
+            warehouse = self.env['stock.warehouse'].browse(self.get_warehouses()[0]['id'])
```

Comment:
I'm thinking maybe you could change this part to directly `search_read` on the `view_location_id` ?
```python
        warehouse_id = self.env.context.get('warehouse', self.get_warehouses()[0]['id'])
        view_location = self.env['stock.warehouse'].search_read([('id', '=', warehouse_id)], ['view_location_id'])
        wh_location_ids = [loc['id'] for loc in self.env['stock.location'].search_read(
            [('id', 'child_of', view_location[0]['id'])],
            ['id'],
        )]
```
The only thing I don't like with this is we could get back the `view_location_id` in the first warehouse's `search_read` (cf. `get_warehouses`) instead of doing a second one, but that's already what we do.

Maybe remove the `get_warehouses` method and call directly a `search_read` instead ? (Here and also in the JS)